app/helpers/i18n_helper.rb

```diff
@@ -47,6 +47,38 @@ def date_eras_gregorian(date)
     nil
   end
 
+  # Return the most accurate date from an array of dates using scoring algorithm below
```

Comment:
Since this method description says the method takes an array of dates and returns the most specific, but the method actually does more than this (it also takes a single non formatted date and formats it; as well as doing the date formatting on the dates) either the comment should be updated, or in order to keep the functionality of this method more atomic to just select the most accurate date, the single string detection/date formatting should be moved into the caller method. Tests would also need to be updated to reflect the fact that this method does NOT take a single string value.